index.js

```diff
@@ -46,32 +48,72 @@ function Filter(inputTree, options) {
   }
 
   this._cache = new Cache();
-  this._canProcessCache = Object.create(null);
-  this._destFilePathCache = Object.create(null);
+
+
+  //newly added
+  this._in = this._out = undefined;
 }
 
+//newly added
+Object.defineProperty(Filter.prototype, 'in', {
+  get() {
+    if (this._in) {
+      return this._in;
+    }
+    this._in = (new FSTree({
+      root: this.inputPaths[0],
+    }));
+    return this._in;
+  }
+});
+
+// should be moved to broccoli-plugin
+Object.defineProperty(Filter.prototype, 'out', {
+  get() {
+    if (this._out) {
+      return this._out;
+    }
+    this._out = (new FSTree({
+      root: this.outputPath
+    }));
+    return this._out;
+  }
+});
+
+
 Filter.prototype.build = function build() {
   var self = this;
   var srcDir = this.inputPaths[0];
```

Comment:
we should be able to get rid of `srcDir` and instead use `this.in[0].resolvePath(relativePath)` when we need it